client/app/project/project.controller.js

```diff
@@ -156,16 +202,100 @@
             var taskPromise = taskService.getTask(projectId, taskId);
             taskPromise.then(function (data) {
                 //task returned successfully
-                vm.selectedTask = data;
+                //vm.selectedTaskFeature = feature;
+                vm.selectedTaskData = data;
                 vm.isSelectTaskMappable = !data.taskLocked && (data.taskStatus === 'READY' || data.taskStatus === 'INVALIDATED');
+                vm.taskError = vm.isSelectTaskMappable ? '' : 'task-not-mappable';
                 vm.currentTab = 'mapping';
-                vm.mappingStep = 'view';
+                vm.mappingStep = 'viewing';
             }, function () {
                 // task not returned successfully
-                vm.selectedTask = null;
+                vm.selectedTaskData = null;
+                //vm.selectedTaskFeature = null;
                 vm.isSelectTaskMappable = false;
                 vm.currentTab = 'mapping';
-                vm.mappingStep = 'task-get-error';
+                vm.taskError = 'task-get-error';
+                vm.mappingStep = 'viewing';
+            });
+        }
+
+        function setLockedTask(task) {
```

Comment:
add comment - also, the function doesn't seem to get used? Or will it get used after the refactor?